extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/client/config/RedisClientConfig.java

```diff
@@ -197,6 +198,29 @@ public interface RedisClientConfig {
     @ConfigDocSection
     TlsConfig tls();
 
+    /**
+     * The client name used to identify the connection.
+     * <p>
+     * If the {@link RedisClientConfig#configureClientName()} is enabled, and this property is not set
+     * it will attempt to extract the value from the {@link RedisClientName#value()} annotation.
+     * <p>
+     * If the {@link RedisClientConfig#configureClientName()} is enabled, both this property and the
+     * {@link RedisClientName#value()} must adhere to the pattern '[^a-zA-Z0-9\\-_.~]', if not adhere,
+     * this may result in an incorrect client name after URI encoding.
+     */
+    Optional<String> clientName();
+
+    /**
+     * Whether it should set the client name while connecting with Redis.
+     * <p>
+     * This is necessary because Redis only accepts {@code client=my-client-name} query parameter after version 6+.
```

Comment:
```suggestion
     * This is necessary because Redis only accepts {@code client=my-client-name} query parameter in version 6+.
```

_After 6+_ seems weird, does that mean 7 only? _In 6+_ seems fine to me.